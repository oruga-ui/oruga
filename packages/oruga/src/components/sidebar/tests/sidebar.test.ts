import { describe, test, expect, afterEach } from "vitest";
import { enableAutoUnmount, mount } from "@vue/test-utils";

import OSidebar from "@/components/sidebar/Sidebar.vue";

describe("OSidebar tests", () => {
    enableAutoUnmount(afterEach);

    test("render correctly", () => {
        const wrapper = mount(OSidebar);
        expect(!!wrapper.vm).toBeTruthy();
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        const element = wrapper.find("div");
        expect(element.attributes("data-oruga")).toBe("sidebar");
    });
});
