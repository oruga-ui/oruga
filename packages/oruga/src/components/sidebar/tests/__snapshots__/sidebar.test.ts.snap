// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OSidebar tests > render correctly 1`] = `
"<!--teleport start-->
<div class="o-side" data-oruga="sidebar">
  <!--v-if-->
  <transition-stub name="slide-prev" appear="false" persisted="true" css="true">
    <div class="o-side__content o-side__content--left o-side__content--hidden" style="display: none;">
      <!--
                        @slot Sidebar default content, default is component prop
                        @binding {(...args):void} close - function to close the component
                    -->
      <!-- injected component for programmatic usage -->
      <!--v-if-->
    </div>
  </transition-stub>
</div>
<!--teleport end-->"
`;
